### PR TITLE
Update CronJobScheduler

### PR DESCRIPTION
**Pull request by jinnnn9**
Update CronJobScheduler such that cron jobs start to run after system is ready. As a reference, it took between 5~10 mins to add all the transaction journal entries in staging.

### CODE REVIEW

**Review by nanoscott — APPROVED** [review]

File: src/foam/nanos/medusa/ClusterConfigSupport.js [hunk 1]
```diff
@@ -806,21 +808,42 @@ configuration for contacting the primary node.`,
...
+          name: 'clusterable',
+          type: 'boolean',
+          documentation: 'true if the cron job is clusterable'
+        }
       ],
       javaCode: `
       try {
+        ElectoralService electoral = (ElectoralService) x.get("electoralService");
+        // System must be ready before running cron jobs
+        if ( electoral.getState() != ElectoralServiceState.IN_SESSION ) {
```
Comment (nanoscott) [inline src/foam/nanos/medusa/ClusterConfigSupport.js]: do you need to check for a null electoral? what will be in the context for local testing?

Comment (jlhughes) [reply src/foam/nanos/medusa/ClusterConfigSupport.js]: @nanoscott This is handled in the cronscheduler - if support == null. 
If support is not null there will be an electoralservice.

Comment (jinnnn9) [reply src/foam/nanos/medusa/ClusterConfigSupport.js]: I think this method is only intended to be invoked when run in a cluster. We always check if ClusterConfigSupport is not null before calling this method . But I don't think the null check would hurt. @nanoscott @jlhughes What do you think?

Comment (jlhughes) [reply src/foam/nanos/medusa/ClusterConfigSupport.js]: Please move this test and the following ! clusterable inside the 
if ( mediator )

**Review by jlhughes — APPROVED** [review]

**Pull request merged**